packages/server/graphql/private/mutations/changeEmailDomain.ts

```diff
@@ -0,0 +1,71 @@
+import {r} from 'rethinkdb-ts'
+import updateDomainsInOrganizationApprovedDomainToPG from '../../../postgres/queries/updateDomainsInOrganizationApprovedDomainToPG'
+import updateUserEmailDomainsToPG from '../../../postgres/queries/updateUserEmailDomainsToPG'
+import {MutationResolvers} from '../../public/resolverTypes'
+
+const changeEmailDomain: MutationResolvers['changeEmailDomain'] = async (
+  _source,
+  {oldDomain, newDomain}
+) => {
+  // VALIDATION
+  if (oldDomain === newDomain) {
+    throw new Error('New domain is the same as the old one')
+  }
+  const normalizedNewDomain = newDomain.toLowerCase()
+  const normalizedOldDomain = oldDomain.toLowerCase()
+
+  // RESOLUTION
+  const [updatedUserIds] = await Promise.all([
+    updateUserEmailDomainsToPG(normalizedOldDomain, normalizedNewDomain),
```

Comment:
After writing the mutation, I noticed we have a `getUsersByDomain` query. I could use `getUsersByDomain` to get the relevant `userIds` and then in `updateUserEmailDomainsToPG` and in the `TeamMember` query, get users with those `userIds` rather than search for matching domains/emails. Do you think it's worth making that change or do you think the current implementation does the job?

Reply:
Thanks for the explanation! I updated the `WHERE` clause but I can see in pgadmin that it's a `SEQ_SCAN`. Same thing happens when `WHERE domain = :oldDomain`. It looks like the scan is happening because of the concatenation which seems to be necessary for the string manipulation. 

![seq-scan](https://user-images.githubusercontent.com/39854876/173047549-249f0f16-627b-432e-ab2b-25298ff16ebc.png)

Reply:
ah, i think i must've confused tables.

`User.domain` is a btree index & will never contain an email, so you can use strict equality (`domain = :oldDomain`)

`OrganizationApproval.domain` is a btree & requires using `LIKE` so it will be a seq scan, which i guess is fine because it'll be a pretty small table.

Reply:
PS, make sure you run that on a table that has a sufficient amount of rows like the staging DB. if there are only 5 rows, it's gonna use a seq scan regardless

<img width="653" alt="Screen Shot 2022-06-10 at 11 05 49 AM" src="https://user-images.githubusercontent.com/5514175/173125235-03b69524-680b-4e1d-b79c-cfd0336ba2dc.png">